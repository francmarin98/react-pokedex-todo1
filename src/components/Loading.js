export const Loading = () => {
    return (
        <div>
            Cargando, espere por favor...
        </div>
    );
};
