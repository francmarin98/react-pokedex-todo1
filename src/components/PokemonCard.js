import {Card, CardContent, CardMedia, Grid, makeStyles, Typography} from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
    cardWrapper: {
        marginTop: "2rem"
    },
    cardMedia: {
        margin: "auto",
    },
    cardContent: {
        textAlign: "center",
    }
}));

export const PokemonCard = ({pokemon, index}) => {
    const classes = useStyles();
    return (
        <Grid item xs={6} sm={3} className={classes.cardWrapper}>
            <Card onClick={() => console.log('Click')}>
                <CardMedia
                    className={classes.cardMedia}
                    image={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${index}.svg`}
                    style={{width: "100px", height: "100px"}}
                />
                <CardContent className={classes.cardContent}>
                    <Typography>{pokemon.name}</Typography>
                </CardContent>
            </Card>
        </Grid>
    );
};
