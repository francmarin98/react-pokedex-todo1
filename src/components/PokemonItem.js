import {Card, CardActionArea, CardContent, CardMedia, Grid, makeStyles, Typography} from "@material-ui/core";
import {useHistory} from "react-router-dom";

const useStyles = makeStyles(() => ({
    cardWrapper: {
        marginTop: "2rem"
    },
    cardMedia: {
        margin: "auto",
        display: "block",
        width: "40%",
        height: "7rem",
        marginTop: "1rem",
        backgroundSize: "contain"
    },
    cardContent: {
        textAlign: "center",
    },
    titleCard: {
        textTransform: "capitalize",
        fontWeight: "bold",
        fontSize: "1rem",
    }
}));

export const PokemonItem = ({pokemon, index}) => {
    const {cardMedia, cardContent, cardWrapper, titleCard} = useStyles();

    const history = useHistory();
    const goToPokemonPage = (name) => history.push(`/pokemon/${name}`);

    return (
        <Grid item xs={6} sm={3} className={`animate__animated animate__fadeIn ${cardWrapper}`}>
            <Card onClick={() => goToPokemonPage(pokemon.name)}>
                <CardActionArea>
                    <CardMedia
                        className={cardMedia}
                        image={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${index}.svg`}
                    />
                    <CardContent className={cardContent}>
                        <Typography variant={"subtitle1"}>
                            {`#${index.toString().padStart(3, 0)}`}
                        </Typography>
                        <Typography className={titleCard}>
                            {pokemon.name}
                        </Typography>
                    </CardContent>
                </CardActionArea>
            </Card>
        </Grid>
    );
};
