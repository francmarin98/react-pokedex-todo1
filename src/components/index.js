export {PokemonMove} from "./PokemonMove/PokemonMove";

export {PokemonDetail} from "./PokemonDetail/PokemonDetail";
export {Header} from "./Header/Header";
export {Footer} from "./Footer/Footer";
export {PokemonItem} from "./PokemonItem";
export {PokemonList} from "./PokemonList";
export {Loading} from "./Loading/Loading";


