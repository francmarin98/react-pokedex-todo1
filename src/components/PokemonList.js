import {Button, Grid, makeStyles} from "@material-ui/core";
import {PokemonItem} from "./PokemonItem";
import {usePokemonesPaginated} from "../hooks";

const useStyles = makeStyles(() => ({
    pokedexContainer: {
        paddingTop: "5rem",
    },
    pagination: {
        margin: " 3rem 0 0 0",
        display: "grid",
        placeItems: "center",
    }
}));

export const PokemonList = () => {
    const {pokedexContainer, pagination} = useStyles();
    const {pokemones, setPage} = usePokemonesPaginated();

    const handleClick = () => setPage(prevPage => prevPage + 1);

    return (
        <>
            <Grid container spacing={2} className={`animate__animated animate__fadeIn ${pokedexContainer}`}>
                {
                    pokemones.map((pokemon, index) => (
                        <PokemonItem
                            key={pokemon.name}
                            pokemon={pokemon}
                            index={index + 1}
                        />
                    ))
                }
            </Grid>
            {
                (pokemones.length !== 0) && (
                    <div className={pagination}>
                        <Button onClick={handleClick} variant="contained" color="primary">
                            Load more
                        </Button>
                    </div>
                )
            }
        </>
    );
};
