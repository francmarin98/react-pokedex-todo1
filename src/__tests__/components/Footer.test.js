import {shallow} from "enzyme";
import {Footer} from "../../components";

describe('Pruebas al componente del footer', () => {
    test('Debe de mostrar el componente', () => {
        const wrapper = shallow(<Footer/>);
        expect(wrapper).toMatchSnapshot();
    })
});
