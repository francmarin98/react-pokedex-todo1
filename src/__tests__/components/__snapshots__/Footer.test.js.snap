// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente del footer Debe de mostrar el componente 1`] = `
<footer
  className="footer"
>
  <a
    href="https://github.com/francmarin98"
    rel="noreferrer"
    target="_blank"
  >
    © 
    2022
     Francisco Marín
  </a>
  <NavLink
    exact={true}
    to="/terms-conditions"
  >
    Terms & Conditions
  </NavLink>
</footer>
`;
